module/event-sourcing/src/Infrastructure/Manager/Decorator/EventStoreManagerCacheDecorator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+/**
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Ergonode\EventSourcing\Infrastructure\Manager\Decorator;
+
+use Ergonode\EventSourcing\Infrastructure\Manager\EventStoreManagerInterface;
+use Ergonode\SharedKernel\Domain\AggregateId;
+use Symfony\Component\Cache\Adapter\AdapterInterface;
+use Ergonode\EventSourcing\Domain\AbstractAggregateRoot;
+
+class EventStoreManagerCacheDecorator implements EventStoreManagerInterface
+{
+    private AdapterInterface $adapter;
+
+    private EventStoreManagerInterface $manager;
+
+    public function __construct(AdapterInterface $adapter, EventStoreManagerInterface $manager)
+    {
+        $this->adapter = $adapter;
+        $this->manager = $manager;
+    }
+
+    public function load(AggregateId $id): ?AbstractAggregateRoot
+    {
+        $item = $this->adapter->getItem($id->getValue());
+        if (!$item->isHit()) {
+            $aggregate = $this->manager->load($id);
+            if ($aggregate) {
+                $item->set($aggregate);
+                $this->adapter->save($item);
+            } else {
+                return null;
```

Comment:
I think we still can save the information that the aggregate is not there? Or you think it makes no sense?

Reply:
fixed